UnitTest_Engine/Query/FilterByCheckStatus.cs

```diff
@@ -0,0 +1,146 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Collections;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.Engine.Serialiser;
+using BH.oM.Base;
+using UT = BH.oM.Test.UnitTests;
+using BH.oM.Test.Results;
+using BH.Engine.Base;
+using BH.oM.UnitTest.Results;
+using BH.Engine.Test;
+using BH.oM.Data.Library;
+using System.IO;
+
+namespace BH.Engine.UnitTest
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Runs and checks all provided UnitTests and provides the ones that has a the provided status.")]
+        [Input("test", "The tests to filter.")]
+        [Input("status", "Status looked for.")]
+        [Input("filterData", "If true, returns new UnitTests objects where the data has been filtered to only contain TestData which returns the provided status. If false, returns full provided UnitTest if the overall status is matching the provided status.")]
+        [MultiOutput(0, "tests", "The filtered tests.")]
+        [MultiOutput(1, "results", "Results matching the filtered tests.")]
```

Comment:
Could we do another output for tests which don't match the filter, so we could daisy chain or inspect the other options from the same component?

Reply:
hum.. Guess we could, but makes things a bit complex with the toggle input, especially when putting it to true. Might be worth a quick chat

Reply:
Went ahead and fixed it